lib/multi_aggregator/data_transfer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module MultiAggregator
+  class DataTransfer
+    attr_reader(
+      :logger
+    )
+
+    def initialize(
+      logger = MultiAggregator::Logger.new
+    )
+      @logger = logger
+    end
+
+    def call(query_spec, storage, providers = {})
+      query_spec.each do |provider_id, tables_spec|
+        transfer_tables(storage, providers[provider_id], provider_id, tables_spec)
```

Comment:
Please let me explain it. I did not have a technical requirements for this gem.
And yes this gem in current state does not suit for big data. But it works for certain purposes.

Could you please specify simple requirements bellow.
1. Spec for sources – types, volumes.
2. Minimal query time execution.
3. TTL for updating data (if it will require cache).
4. Types of query or examples.
5. Other important information.

After it I will try to prepare solution for your requirements.

PS. We have postresql and redis. Both are good products for certain purposes.
Please do not ask me to find silver bullet ;)